openedx/core/djangoapps/content/learning_sequences/models.py

```diff
@@ -111,7 +111,16 @@ class LearningSequence(TimeStampedModel):
     learning_context = models.ForeignKey(
         LearningContext, on_delete=models.CASCADE, related_name='sequences'
     )
-    usage_key = UsageKeyField(max_length=255)
+    usage_key = UsageKeyField(max_length=255, unique=True)
+
+    # A URL-safe Base64-encoding of a blake2b hash of the usage key.
+    # For aesthetic use (eg, as a path parameter, to shorten URLs).
+    # This field is experimental. See TNL-8638 for more information.
+    usage_key_hash = models.CharField(
+        max_length=USAGE_KEY_HASH_LENGTH,
+        null=True,
+        default=None,
+    )
```

Comment:
This needs an index of its own, or the API query that fetches using it will have to do a full table scan.

Reply:
Didn't realize the order made a difference. I'll swap the order of those.

Reply:
Yeah, it doesn't matter for the purposes of enforcing the constraint, but it does for speed of lookup. When you have a composite index of `(a, b)`, the database can use it for lookups on just `a` as well.